### ISSUE TITLE
Command option to change font size

### ISSUE THREAD

**Issue by phuchptty** [ToaHartor/GI-cutscenes#41]
like the title, I hope the feature change font size because the subtitle is still a bit large.

I will try to look into it when I have time

**Comment by ToaHartor** [comment]
Good idea, this can be included in the CLI improvements.

In fact, it's quite easy to do at the first set conversion, by modifying the ASS header in the generation. However, if the ASS already exists, then we will have to modify the default style in the existing ASS.

**Comment by phuchptty** [comment]
I think we can modify it in the engine struct. Bypassing an argument to the engine.
It will override the ASS default setting, I guess

**Comment by ToaHartor** [comment]
Tried to look at it quickly, and all I could find is the modification of the subtitles file, for both FFMPEG and mkvmerge. I guess that's mainly because of the way the mkv format works. 
In theory, the subtitles in a mkv file is copied in raw, and there is no tag to override the style inside. As the subtitles aren't burned in the video, it's up to the player to read them. Therefore, it's possible to modify the font style and size during the playback. But if we want to impose a font style (with the official font) and the right size, it has to be done directly in the ASS file.

[This line in ASS.cs](https://github.com/ToaHartor/GI-cutscenes/blob/main/src/FileTypes/ASS.cs#L82), right after the `_fontname` is the fontsize. Like the font name, we can add the font size directly at the object creation, and then reuse it in `ConvertToAss()` and in a new function that would parse an ASS file and modify the line starting with `Style: Default,{_fontname},`.

**Comment by phuchptty** [comment]
Okay, I see. But we should fix the issue with duplicate subfile (txt and srt) first. After that adding font size is pretty simple.

**Comment by Scighost** [comment]
I think the following subtitle styles are more appropriate. There is still a little difference in shadow and text spacing, but it doesn't matter.

Change `Fontsize` to 12 and `MarginV` to 14.

``` diff
- Style: Default,{_fontname},18,&H00FFFFFF,&H000000FF,&H00000000,&H00000000,0,0,0,0,100.0,100.0,0.0,0.0,1,0,0.5,2,10,10,20,1
+ Style: Default,{_fontname},12,&H00FFFFFF,&H000000FF,&H00000000,&H00000000,0,0,0,0,100.0,100.0,0.0,0.0,1,0,0.5,2,14,14,14,1
```

Comparison:

![未标题-1](https://user-images.githubusercontent.com/61003590/197715316-3e344e39-38ca-41d8-a342-9fbb15573824.png)

**Comment by phuchptty** [comment]
Yeah. But there is a problem is when you convert subtitles to ass. It will cache in your subtitles folder, so I'm finding some way to remove it if the font size change

**Comment by Scighost** [comment]
> Yeah. But there is a problem is when you convert subtitles to ass. It will cache in your subtitles folder, so I'm finding some way to remove it if the font size change

Maybe that's your video player's problem.

**Comment by Scighost** [comment]
I see.
The subtitles in repo `GenshinData` only have `srt` and `txt` file without style. When I use `--subs` option, the program will convert `srt` to `ass`. It's a bad way.
I think the program cannot modify any file in GenshinData folder, and the converted ass files can be saved in a temp folder.

**Comment by ToaHartor** [comment]
Thinking about it, yes that's not really convenient. An idea would be to create a "subs" folder with a folder structure similar to the `GenshinData` repository. The program could then first check this folder to get the .ass files, then if it doesn't find anything, fallback to the GenshinData folder. If there is any modification to do to the files, it could then directly modify the .ass files in the right folder.

Would it require to have an option in the command line/settings file to indicate the path of it to make it configurable to the user ?

**Comment by Scighost** [comment]
Create `subs` folder in `output` folder is a better way, and if user not add `--no-cleanup` option, delete `subs` folder at the end. GenshinData doesn't delete any subtitle file since first commit, so it's unnecessary to reserve `.ass` file.

If user wants to custom one video subtitle style, he needs to modify at least 13 files manually and tediously. A solution is define a optional `SubsStyle` key in `appsetting.json`, if the key not be changed, fallback to the default style.

For example:

``` json
"SubsStyle": "Style: Default,{_fontname},12,&H00FFFFFF,&H000000FF,&H00000000,&H00000000,0,0,0,0,100.0,100.0,0.0,0.0,1,0,0.5,2,14,14,14,1"
```
